src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -256,22 +250,30 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                 // CRLF for end of headers.
                 await WriteTwoBytesAsync((byte)'\r', (byte)'\n', cancellationToken).ConfigureAwait(false);
 
-                // Write body, if any
+                // Flush out anything buffered from the request headers to finish sending them.
+                await FlushAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Yeah, it's not crazy.  The problem is, I don't know how to make this work while keeping the "headers and initial write in a single send" optimization.